src/components/views/rooms/RoomSublist.tsx

```diff
@@ -522,11 +523,13 @@ export default class RoomSublist extends React.Component<IProps, IState> {
             const visibleRooms = this.state.rooms.slice(0, this.numVisibleTiles);
             for (const room of visibleRooms) {
                 tiles.push(<RoomTile
-                    room={room}
+                    model={new RoomTileViewModel({
```

Comment:
That might be personal preferences, but I believe spreading the enumerable properties of the view model onto the component might be better

React only does shallow comparison on props for things like `React.memo`. Which would force us to implement a `areEqual` function every time 
It also reads a bit better when going through the debugger sidebar for example

Reply:
The same goes for the props comparison with `React.PureComponent`. We'd have to implement `shouldComponentUpdate`

Reply:
Not quite sure what my preference would be yet, but about `shouldComponentUpdate`, I believe you'd need either way when there are complex nested objects inside the model, such as `room`.

Reply:
Yeah, spreading the view model properties was also my first though. I haven't come up with a good way to trigger the component to rerender though this way. The parent would have to rerender basically when a property on one of the child view models changes. And what triggers the parent to rerender if not forceUpdate or some state variable? Might be better to just put that in the child component then and not also rerender all the sibling components.

Reply:
I'd be happy to learn about other possibilities here though

Reply:
I actually went with spreading the view model properties over the component state (so it can update itself by listening to the view model)